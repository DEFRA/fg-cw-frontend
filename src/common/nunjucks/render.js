import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
