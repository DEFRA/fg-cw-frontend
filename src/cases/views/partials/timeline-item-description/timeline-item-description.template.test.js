import { describe, expect, it } from "vitest";
import { render } from "../../../../common/nunjucks/render.js";

describe("timeline-item-description", () => {
  it("renders case assigned view", () => {
    const component = render("timeline-item-description", {
      eventType: "CASE_ASSIGNED",
      commentRef: "12334-POPOP",
      caseId: "9999-00000",
      data: {
        assignedTo: { name: "Mickey Mouse" },
      },
    });
    expect(component).toMatchSnapshot();
  });

  it("renders other view", () => {
    const component = render("timeline-item-description", {
      caseId: "9999-00000",
      eventType: "NOTE_ADDED",
      commentRef: "comment-12345",
    });

    expect(component).toMatchSnapshot();
  });
});
