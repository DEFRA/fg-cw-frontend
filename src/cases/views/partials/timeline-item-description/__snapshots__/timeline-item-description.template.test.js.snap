// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`timeline-item-description > renders case assigned view 1`] = `
"<div class="timeline__description">
    <p>
      Assigned to Mickey Mouse
    </p>
</div>
"
`;

exports[`timeline-item-description > renders other view 1`] = `
"<div class="timeline__description">
    <!-- TODO  note link? -->
    <a class="govuk-link" href="/notes">View note</a>
</div>
"
`;
